resources/default.rb

```diff
@@ -0,0 +1,133 @@
+property :link_name, String, name_property: true
+property :link, String
+property :path, String
+property :priority, [String, Integer]
+
+action :install do
+  alternatives_install
+end
+
+action :set do
+  alternatives_set
+end
+
+action :remove do
+  alternatives_remove
+end
+
+action :auto do
+  alternatives_auto
+end
+
+action :refresh do
+  alternatives_refresh
+end
+
+action_class do
+  def alternatives_cmd
+    if platform_family?('rhel', 'amazon', 'fedora')
+      'alternatives'
+    else
+      'update-alternatives'
+    end
+  end
+
+  def link_path
+    new_resource.link ? new_resource.link : "/usr/bin/#{new_resource.link_name}"
+  end
+
+  def validate_path
+    unless new_resource.path
+      raise "could not set alternatives for #{new_resource.link_name}, must provide :path"
+    end
+    unless ::File.exist?(new_resource.path)
+      raise "could not set alternatives for #{new_resource.link_name}, missing #{new_resource.path}"
+    end
+  end
```

Comment:
this should go into define_resource_requirements

Reply:
Done